src/cswinrt/strings/WinRT.cs

```diff
@@ -450,48 +453,75 @@ public static unsafe (ObjectReference<IActivationFactoryVftbl> obj, int hr) GetA
-
-        public ObjectReference<IActivationFactoryVftbl> Value { get => _IActivationFactory; }
-
-        public I AsInterface<I>() => _IActivationFactory.AsInterface<I>();
-
-        public BaseActivationFactory(string typeNamespace, string typeFullName)
+        private readonly IntPtr _contextToken;
+
+        public static FactoryObjectReference<T> Attach(ref IntPtr thisPtr)
         {
-            // Prefer the RoGetActivationFactory HRESULT failure over the LoadLibrary/etc. failure
-            int hr;
-            (_IActivationFactory, hr) = WinrtModule.GetActivationFactory(typeFullName);
-            if (_IActivationFactory != null) { return; }
-
-            var moduleName = typeNamespace;
-            while (true)
+            if (thisPtr == IntPtr.Zero)
             {
-                DllModule module = null;
-                if (DllModule.TryLoad(moduleName + ".dll", out module))
-                {
-                    (_IActivationFactory, _) = module.GetActivationFactory(typeFullName);
-                    if (_IActivationFactory != null) { return; }
-                }
-
-                var lastSegment = moduleName.LastIndexOf(".", StringComparison.Ordinal);
-                if (lastSegment <= 0)
-                {
-                    Marshal.ThrowExceptionForHR(hr);
-                }
-                moduleName = moduleName.Remove(lastSegment);
+                return null;
+            }
+            var obj = new FactoryObjectReference<T>(thisPtr);
+            thisPtr = IntPtr.Zero;
+            return obj;
+        }
+
+        internal FactoryObjectReference(IntPtr thisPtr) :
+            base(thisPtr)
+        {
+            if (!IsFreeThreaded(this))
+            {
+                _contextToken = Context.GetContextToken();
+            }
+        }
+
+        internal FactoryObjectReference(IntPtr thisPtr, IntPtr contextToken)
+            : base(thisPtr)
+        {
+            _contextToken = contextToken;
+        }
+
+        public static new unsafe FactoryObjectReference<T> FromAbi(IntPtr thisPtr)
+        {
+            if (thisPtr == IntPtr.Zero)
+            {
+                return null;
             }
+            var obj = new FactoryObjectReference<T>(thisPtr);
+            obj.VftblIUnknown.AddRef(obj.ThisPtr);
+            return obj;
         }
 
-#if NET
-        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2091:RequiresUnreferencedCode",
-            Justification = "No members of the generic type are dynamically accessed in this code path.")]
-#endif
-        public unsafe ObjectReference<I> _ActivateInstance<I>()
+        public bool IsObjectInContext()
+        {
+            return _contextToken == IntPtr.Zero || _contextToken == Context.GetContextToken();
+        }
+
+        // If we are free threaded, we do not need to keep track of context.
+        // This can either be if the object implements IAgileObject or the free threaded marshaler.

```

Comment:
I assume we are not checking the free threaded marshaler as it is internal to WinRT.Runtime.  We can probably address this in the future, but we should update the comment to reflect that.